test/integration/resourcemanager/managedresource/resource_test.go

```diff
@@ -985,6 +985,49 @@ var _ = Describe("ManagedResource controller tests", func() {
 			})
 		})
 	})
+
+	Describe("Immutable resources", func() {
+		It("should fail to apply the resource if the update is invalid", func() {
+			newData := map[string]string{
+				"foo": "bar",
+			}
+
+			Eventually(func(g Gomega) []gardencorev1beta1.Condition {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+				return managedResource.Status.Conditions
+			}).Should(
+				ContainCondition(OfType(resourcesv1alpha1.ResourcesApplied), WithStatus(gardencorev1beta1.ConditionTrue), WithReason(resourcesv1alpha1.ConditionApplySucceeded)),
+			)
+
+			Expect(testClient.Delete(ctx, configMap)).To(Succeed())
+
+			configMap.Data = newData
+			configMap.Immutable = pointer.Bool(true)
```

Comment:
Nit: We know that for reproducing the issue it is not required the initial ConfigMap to be immutable. But does it make sense to make it immutable to be closer to the real world?